### ISSUE TITLE
[1.12.2]Massive memory leak with mystcraft(possibly other dimension mods to)

### ISSUE THREAD

**Issue by Will11690** [Azanor/thaumcraft-beta#1659]
While playing a a private server with a friend I noticed that if one of us was in the overworld and the other was in a mystcraft dimension the servers ram usage would keep increasing rapidly and eventually crash the server. When I check the console this is what I found.

[11:48:30] [Server thread/INFO] [THAUMCRAFT]: Creating aura cache for world 7
[11:48:30] [Server thread/INFO] [THAUMCRAFT]: Removing aura cache for world 7
[11:48:30] [Server thread/INFO] [FML]: Unloading dimension 7
[11:48:30] [Server thread/INFO] [FML]: Loading dimension 7 (NewWorld) (net.minecraft.server.dedicated.DedicatedServer@6e580501)

It happened over and over again very rapidly and sent us from 2GB of usage to 5 before I noticed it.

**Comment by Will11690** [comment]
removing thaumcraft from the pack fixed the issue

**Comment by ghost** [comment]
This appears to occur within my modpack also.

**Comment by ghost** [comment]
@Azanor You will seriously need to take a look at this.

**Comment by Will11690** [comment]
I just found out recently Azanor has retired once again for various reasons. So basically this mod is dead until/if he comes back

**Comment by Serifina** [comment]
Then you also missed that a team, headed by KingLemming, will be maintaining and updating the mod.

**Comment by ghost** [comment]
Please further forward this issue to them.
Don't mean to rush y'all but this is quite a serious issue as:

- This severely affects loading times into Singleplayer and initial Server Starting.
- This causes persistent freezing throughout the game. Making it a terrible experience.

**Comment by Redcateth** [comment]
Can confirm this is an issue when creating my modpack as well. Does anyone know whether this is an issue on any of the older versions of Thaumcraft or if this is being caused by newer versions of Forge?
Edit: Issue isn't confined to Mystcraft itself, seems to be a problem with other dimensions in general.

**Comment by Redcateth** [comment]
Alright, temporary solution: Problem seems to have something to do with how the newer versions of Forge handle dimensions. Went from version forge-14.23.5.2835 all the way back to forge-14.23.5.2800 and it seems to have fixed the issue. Could probably use a more updated version than that but I haven't tested that far yet.

**Comment by ItsMestro** [comment]
Want to chime in with my solutions to this problem when I encountered it in one of my worlds.
I'm currently playing the modpack Project Ozone 3 Beta 3.0.40 running forge 14.23.5.2838 with thaumcraft, thaumic tinkerer and Ex Compressum added by myself. For the first couple of hours playing the modpack I didn't encounter any problem but after beginning to do some thaumcraftery I started having this memory leak problem. Game output log was constantly showing messages about loading and unloading worlds/aura cache.
After spending about 6 hours debugging it I found it to boil down to chunk loading or anything that updates information in a chunk like entity counts etc. Anything that does this for a chunk where any thaumcraft block is placed will create this problem. In my case I had the crucible, research table and crafting bench in the same chunk as a running mob duplicator from Industrial Foregoing.
Fixing this problem was as simple as stopping the duplicator. Moving it to a different chunk also proved to work. The problem and solution stayed the same with all the versions of forge, IF and TC that I was able to try with this modpack. Replacing the duplicator with different chunkloaders had the same effect on the game.

**TL;DR**
**IF** you get this problem on a map that you've played for a little while.
**THEN** check to make sure you don't have any chunkloaders or similar things in chunks with thaumcraft blocks/structures.
